### PR TITLE
loadMemberContext() loads data indiscriminately

### PR DESCRIPTION
**Pull request by ghost**
-  Set a context var to pass the $set value
-  Split the $memberContext array between the always loaded values and those loaded when using "normal"

I realize this is a big change that might had lots of consequences, still, just a possible fix for [bug 5017]

### CODE REVIEW

**Comment by emanuele45** [comment]
I think it makes sense (but for this evening I already merged enough, I'll bring this in tomorrow).

**Comment by live627** [comment]
I don't see the harm in this.

File: Sources/Load.php [hunk 1]
```diff
@@ -876,12 +876,15 @@ function loadPermissions()
  */
 function loadMemberData($users, $is_name = false, $set = 'normal')
```
Comment (norv) [inline Sources/Load.php:877]: If there is only 'normal' and 'minimal' intended, then a boolean is simpler. i.e.

``` php
function loadMemberData($users, $is_name = false, $minimal = false)
```

**Comment by ghost** [comment]
Theres also "profile".  I've been looking at it more, apart from this changes I was thinking on making edits to how $select_columns and $select_tables are filled, theres a lot of redundant data and those ifs can be changes to a switch ith the default value been the trigger_error func.

**Comment by ghost** [comment]
I'm also thinking about those vars ($select_columns and $select_tables) and the hook, I'm actually thinking on convert those strings to an array, pass the array to the hook as an array is much more easier to handle than a string and then when we actually are using those var we can do an implode() to revert back.

Also as ema said to me, passing the set var to the hook might be a good idea.

Edit, the IFNULL THEN and ELSE things can be a problem if changed to an array... mmm, dunno whats the best thing to do here :(

**Comment by live627** [comment]
Do it.

Why would the IF statements be a problem? Just include the entire block in an array value.

File: Sources/Load.php (hunk 1 above)
Comment (live627) [inline Sources/Load.php:877]: Won't cut it, sorry. The profiles will break... hard..

Comment (norv) [inline Sources/Load.php:877]: You folks are right, it won't, as MAS noticed already. My mistake. I blame lack of coffee, and emanuele by default. (nobody needs a reason to blame Ema.)

**Pull request merged**